tutorials/xr/setting_up_xr.rst

```diff
@@ -0,0 +1,107 @@
+-------------------------------
+
+While in Godot 3 most things worked out of the box, Godot 4 needs a little more setup. This is mainly due to the more advanced nature of the Vulkan renderer. There are many rendering features in Vulkan the XR system uses that aren't enabled by default. They are very easy to turn on, simply open up your project settings and tick the XR shaders tickbox in the XR section:
+
+.. image:: img/xr_shaders.png
+
+.. warning::
+	As Godot 4 is still in development, many post process effects have not yet been updated to support stereoscopic rendering. Using these will have adverse effects.
+
+.. note::
+	Godot also has the choice between a desktop and mobile Vulkan renderer. There are a number of optimisations added to the mobile renderer that benefit XR applications. You may wish to enable this even on desktop.
+
+OpenXR
+------
+
+OpenXR is a new industry standard that allows different XR platforms to present themselves through a standardised API to XR applications. This standard is an open standard maintained by the Khronos Group and thus aligns very well with Godots interests.
+
+The Vulkan implementation of OpenXR is closely integrated with Vulkan, taking over part of the Vulkan system. This requires tight integration of certain core graphics features in the Vulkan renderer which are needed before the XR system is setup. This was one of the main deciding factors to include OpenXR as a core interface.
+
+This also means OpenXR needs to be enabled when Godot starts in order to set things up correctly. The required setting can be found in your project settings:
+
+.. image:: img/openxr_settings.png
+
+As setup is brought forward with OpenXR you can find several other settings related to OpenXR here as well. These can't be changed after your application is running.
+The default settings will get us started and we will go into detail in another section of the documentation.
+
+Setting up the XR scene
+-----------------------
+
+Every XR application needs at least an :ref:`XROrigin3D <class_xrorigin3d>` and an :ref:`XRCamera3D <class_xrcamera3d>` node. Most will have two :ref:`XRController3D <class_xrcontroller3d>`, one for the left hand and one for the right. Keep in mind that the camera and controller nodes should be children of the origin node. Add these nodes to a new scene and rename the controller nodes to ``LeftHand`` and ``RightHand``, your scene should look something like this: 
+
+.. image:: img/xr_basic_scene.png
+
+Next you need to configure the controllers, select the left hand and set it up as follows:
+
+.. image:: img/xr_left_hand.png
+
+And the right hand:
+
+.. image:: img/xr_right_hand.png
+
+Right now all these nodes are on the floor, the will be positioned correctly in runtime. To help during development I like to move the camera upwards so its ``y`` is set to ``1.7``, and move the controller nodes to ``-0.5, 1.0, -0.5`` and ``0.5, 1.0, -0.5`` for respectively the left and right hand.
```

Comment:
```suggestion
Right now all these nodes are on the floor, the will be positioned correctly in runtime. To help during development, it can be helpful to move the camera upwards so its ``y`` is set to ``1.7``, and move the controller nodes to ``-0.5, 1.0, -0.5`` and ``0.5, 1.0, -0.5`` for respectively the left and right hand.
```